.github/workflows/check-links.yaml

```
@@ -17,35 +17,44 @@ jobs:
   changedfiles:
     name: changed files
     runs-on: ubuntu-latest
-    env:
-      PR_HEAD: ${{ github.event.pull_request.head.sha }}
     if: ${{ github.actor != 'dependabot[bot]' }}
     outputs:
-      md: ${{ steps.changes.outputs.md }}
+      md_files: ${{ steps.changed-files.outputs.md_all_changed_files }}
+      yaml_files: ${{ steps.changed-files.outputs.yaml_all_changed_files }} # used to catch MD_LINK_CHECK_VERSION updates and runs check-links to confirm functionality
     steps:
       - uses: actions/checkout@v4
         with:
           fetch-depth: 0
       - name: Get changed files
-        id: changes
-        run: |
-          echo "md=$(git diff --name-only --diff-filter=ACMRTUXB $(git merge-base origin/main $PR_HEAD) $PR_HEAD | grep .md$ | xargs)" >> $GITHUB_OUTPUT
+        id: changed-files
+        uses: tj-actions/changed-files@v45
+        with:
+          files_yaml: |
+            md:
+              - '**.md'
+            yaml:
+              - .github/workflows/check-links.yaml
+
   check-links:
     runs-on: ubuntu-latest
     needs: changedfiles
-    if: ${{needs.changedfiles.outputs.md}}
+    if: needs.changedfiles.outputs.md_files || needs.changedfiles.outputs.yaml_files 
     steps:
       - uses: actions/checkout@v4
         with:
           fetch-depth: 0
-
       - name: Install markdown-link-check
         run: npm install -g markdown-link-check@${{ env.MD_LINK_CHECK_VERSION }}
-
       - name: Run markdown-link-check
         run: |
+          if [ -z "${{ needs.changedfiles.outputs.md_files }}" ]; then
+            md="./*.md"
```

wouldn't this need to be a `**` glob to go into subdirs?
```suggestion
            md="./**.md"
```

---

I'm somehow torn between checking all markdown files or only the ones in the top-level directory. I tested to scan all files and it took approx 10min to succeed then: https://github.com/niwoerner/opentelemetry-collector-contrib/actions/runs/12175439720/job/33959187810 

On one side it won't hurt (and might even show broken links we otherwise would miss) but on the other side I felt like 10min is a bit much and given the growing amount of markdown files in the repo here - it might fail frequently due to some broken links. 

But nevertheless I'm completely fine with changing it, what do you think? @mowies

---

I would change it in this case and also consider sub-directories. The runtime is not super important IMO, we have other runs that take much longer, so PRs would probably not be generally blocked for a longer time with this. (I also think that the org in general has enough CI minutes to spare, but this is an educated guess at best)

Also, I think it makes sense to check all markdown files once the link checker version updates. That is the only way to make sure that a new change in the link checker (and potential needed config of markdown updates) is caught when the actual version bump happens (through renovate) and not in some unrelated PRs down the line where somebody unknowing of this changes something in a markdown file.